foreman/data_refinery_foreman/foreman/management/commands/rerun_salmon_old_samples.py

```diff
@@ -18,15 +18,21 @@
 
 
 def update_salmon_versions(experiment: Experiment):
-    quant_results = (
-        get_quant_results_for_experiment(experiment, filter_old_versions=False)
-        .order_by("-organism_index__created_at")
-        .prefetch_related("organism_index")
-        .prefetch_related("samples__original_files")
-    )
+    quant_results = get_quant_results_for_experiment(experiment, filter_old_versions=False)
+
+    # We first need to find the last created result and get its salmon version.
+    # get_quant_results_for_experiment returns a set, not a queryset.
+    last_created = None
+    for quant_result in quant_results:
+        if not last_created:
+            last_created = quant_result
+        else:
+            if quant_result.created_at > last_created.created_at:
+                last_created = quant_result
+
+    latest_salmon_version = last_created.organism_index.salmon_version
 
     total_samples_queued = 0
-    latest_salmon_version = None
     for quant_result in quant_results:
         if not latest_salmon_version:
             # we can safely ignore the latest salmon version, that will be the first
```

Comment:
I don't think the comment here still applies, because we are no longer ordering by `-organism_index__created_at`. This branch also might be unreachable because `latest_salmon_version` is already set before this loop.

Reply:
Good catch, this is dead code that should have come up when I added the `last_created` loop.